spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/H2ServerConfiguration.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.config;
+
+import java.sql.SQLException;
+
+import org.h2.tools.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Autostart an embedded H2 database server.
+ *
+ * @author Michael Wirth
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnClass(Server.class)
+@ConditionalOnProperty(value = "spring.dataflow.embedded.database.enabled", havingValue = "true", matchIfMissing = true)
+@ConditionalOnExpression("'${spring.datasource.url:#{null}}'.startsWith('jdbc:h2:tcp://localhost:')")
+public class H2ServerConfiguration {
+
+	private static final Logger logger = LoggerFactory.getLogger(H2ServerConfiguration.class);
+
+	@Bean
+	public H2ServerBeanFactoryPostProcessor h2ServerBeanFactoryPostProcessor() {
+		return new H2ServerBeanFactoryPostProcessor();
+	}
+
+	@Bean
+	public Server initH2TCPServer(@Value("${spring.datasource.url}") String dataSourceUrl) {
+		logger.info("Starting H2 Server with URL: " + dataSourceUrl);
+		try {
+			Server server = Server.createTcpServer("-ifNotExists", "-tcp",
+					"-tcpAllowOthers", "-tcpPort", getH2Port(dataSourceUrl)).start();
+			server.setShutdownHandler(server::stop);
+			return server;
+		}
+		catch (SQLException e) {
+			throw new IllegalStateException(e);
+		}
+	}
+
+	private String getH2Port(String url) {
```

Comment:
It may be cleaner to just inline this now that its only called in a single place.

Reply:
I inlined it and replaced it with a Regex expression

Reply:
Looks good. I will try to not suggest that we should take advantage of SpEL support for regex and `matches` operation. Because the next iteration of that would be to share the pattern which would make the SpEL very "not pretty" :)